subscriptions.go

```diff
@@ -101,6 +133,24 @@ func (s *SubscriptionAPI) List() ([]*Subscription, error) {
 	return subscriptions.Items, nil
 }
 
+// ListFilter returns all subscriptions that consistent with the `SubscriptionParameter`s
+// DEPRECATED: will be replaced in version 2. Use `List()` instead
+func (s *SubscriptionAPI) ListFilter(params SubscriptionParameters) ([]*Subscription, error) {
```

Comment:
The method `ListFilter` should have a unit test (`subscriptions_test.go`) and a integration test (`subscriptions_integration_test.go`). For the integration test it is OK to just test the happy path.